packages/core/src/store/actions.ts

```diff
@@ -140,6 +156,100 @@ export function updateAccountCenter(
   dispatch(action as UpdateAccountCenterAction)
 }
 
+export function updateNotify(update: Partial<NotifyOptions>): void {
+  const error = validateNotifyOptions(update)
+
+  if (error) {
+    throw error
+  }
+
+  const action = {
+    type: UPDATE_NOTIFY,
+    payload: update
+  }
+
+  dispatch(action as UpdateNotifyAction)
+}
+
+export function addNotification(notification: Notification): void {
+  const error = validateNotification(notification)
+
+  if (error) {
+    throw error
+  }
+
+  const action = {
+    type: ADD_NOTIFICATION,
+    payload: notification
+  }
+
+  dispatch(action as AddNotificationAction)
+}
+
+export function addCustomNotification(notification: CustomNotification): void {
+  const customNotificationError = validateCustomNotification(notification)
+
+  if (customNotificationError) {
+    throw customNotificationError
+  }
+
+  const action = {
+    type: ADD_NOTIFICATION,
+    payload: notification
+  }
+
+  dispatch(action as AddNotificationAction)
+}
+
+export function customNotification(updatedNotification: Omit<CustomNotification, 'id' | 'key'>): {
+  dismiss: () => void
+  update: UpdateNotification
+} {
+  
+  const customIdKey = `customNotification-${nanoid()}`
+  updatedNotification.id = customIdKey
+  updatedNotification.key = customIdKey
+  addCustomNotification(updatedNotification)
+
+  const dismiss = () => removeNotification(updatedNotification.id)
+  
+  const update = (
+    notificationUpdate: Omit<CustomNotification, 'id' | 'key'>
```

Comment:
It might be better to change the definition of `CustomNotification` type to omit the `id` and `key` and then use that since we don't really want the `id` or `key` to be changed when customizing from the transaction handler either.

Reply:
Cool, I made a new type to handle this and enforce more structure